softioc/device_core.py

```diff
@@ -126,8 +126,7 @@ class DSET(DSET_BASE):
         # Hang onto the values we publish to EPICS to ensure that they persist!
         # We also need to ensure that the device name persits.
         cls.__dset = dset
-        if sys.version_info >= (3,):
-            cls._device_name_ = cls._device_name_.encode()
+        cls._device_name_ = cls._device_name_.encode()
         imports.registryDeviceSupportAdd(cls._device_name_, byref(cls.__dset))
```

Comment:
Looking at how `_device_name_` is used, it's probably cleaner to rewrite these two lines as:
```python
       imports.registryDeviceSupportAdd(
            cls._device_name_.encode(), byref(cls.__dset))
```
The only reason this string was encoded was precisely for this method code.

Even better, how about changing the first field of `argtypes` for `registerDeviceSupportAdd` (in `imports.py`) from `c_char_p` to `auto_encode`?  Then we don't need to call `.encode()` at all!

Reply:
As discussed, we cannot do this as Python must keep hold of the data, to ensure the memory is not garbage-collected.

Reply:
Yep :)